pennylane/ops/op_math/exp.py

```diff
@@ -154,10 +169,15 @@ class Exp(ScalarSymbolicOp, Operation):
     control_wires = Wires([])
     _name = "Exp"
 
-    def __init__(self, base=None, coeff=1, do_queue=True, id=None):
+    # pylint: disable=too-many-arguments
+    def __init__(self, base, coeff=1, num_steps=None, do_queue=True, id=None):
         super().__init__(base, scalar=coeff, do_queue=do_queue, id=id)
         self._data = [[coeff], self.base.data]
         self.grad_recipe = [None]
+        self.num_steps = num_steps
+        self._ops_with_generator = [
+            getattr(qml, op_name) for op_name in has_unitary_generator if op_name != "PauliRot"
+        ]
```

Comment:
Do we need to dynamically recalculate this upon initialization? Is this to accomodate someone adding a new operator to `has_unitary_generator` after initializing PL?

And why doesn't PauliRot count?

Reply:
> Do we need to dynamically recalculate this upon initialization?

Yes, otherwise we'd get circular import errors... Ideally we should use class types (instead of strings) in all PL attributes.

Edit: I just realised we can move this list inside the attributes file, and build it only once.

> And why doesn't PauliRot count?

Because the initialisation is different from other operations (it requires a string containing the pauli word). Would be nice to support operators too.

Reply:
> Edit: I just realised we can move this list inside the attributes file, and build it only once.

Done!